e2e_tests/tests/api_utils.py

```diff
@@ -324,6 +333,42 @@ def decorator(f: F) -> F:
     return decorator
 
 
+def skipif_unexpected_master_config(
+    expected: Callable[[dict], bool], reason: str = "unexpected master config"
+) -> Callable[[F], F]:
+    def decorator(f: F) -> F:
+        mc = _get_master_config()
+        if mc is None:
+            return f
+        if not expected(mc):
+            return pytest.mark.skipif(True, reason=reason)(f)  # type: ignore
+        return f
+
+    return decorator
+
+
+def skipif_missing_startup_hook(
+    reason: str = "tcd startup hook is required for this test",
+) -> Callable[[F], F]:
+    """skip if the backing cluster can be missing TCD startup hooks for some of the workloads"""
+
+    def check_startup_hook(mc: dict) -> bool:
+        assert isinstance(mc, dict)
+
+        def has_hook(conf_with_tcd: dict) -> bool:
+            hook = (conf_with_tcd.get("task_container_defaults") or {}).get("startup_hook")
+            return isinstance(hook, str) and hook != ""
+
+        if has_hook(mc):
+            return True
+        pools = mc.get("resource_pools", [])
+        for p in pools:
+            print(p)
```

Comment:
shoot I forgot to take this out but not gonna rerun ci at this point since we're waiting to cherry-pick it